src/passes/ConstantFieldPropagation.cpp

```diff
@@ -210,7 +210,7 @@ struct FunctionOptimizer : public WalkerPass<PostWalker<FunctionOptimizer>> {
     // on simply applying a constant. However, we can try to use a ref.test, if
     // that is allowed.
     if (!info.isConstant()) {
-      if (refTest) {
+      if (refTest && !ref->type.isExact()) {
```

Comment:
Maybe add a comment here, e.g. `Note that if the reference is exact, we never need to use a ref.test, as we will not have two types to select between`